File: arrays-hashing/med-0128-longest-consecutive-sequence/Solution.java
import java.util.Arrays;

class Solution {
    public int longestConsecutive(int[] nums) {
        if (nums.length == 0) return 0;

        Arrays.sort(nums);

        int maxCount = 1;
        int currCount = 1;

        for (int i = 1; i < nums.length; i++) {
            if (nums[i] == nums[i - 1]) {
                continue;
            } else if (nums[i] == nums[i - 1] + 1) {
                currCount++;
            } else {
                maxCount = Math.max(maxCount, currCount);
                currCount = 1;
            }
        }

        maxCount = Math.max(maxCount, currCount);
        return maxCount;
    }
}

File: arrays-hashing/med-0128-longest-consecutive-sequence/Solution.py
from typing import List

class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if not nums:
            return 0

        nums.sort()

        max_count = 1
        curr_count = 1

        for i in range(1, len(nums)):
            if nums[i] == nums[i - 1]:
                continue
            elif nums[i] == nums[i - 1] + 1:
                curr_count += 1
            else:
                max_count = max(max_count, curr_count)
                curr_count = 1

        max_count = max(max_count, curr_count)
        return max_count
